tests/driver_bme680/main.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2018 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     tests
+ * @{
+ *
+ * @file
+ * @brief       Test application for the bme680_driver package.
+ *
+ * @author      Dylan Laduranty <dylan.laduranty@mesotic.com>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+#include "board.h"
+#include "bme680.h"
+#include "bme680_params.h"
+#include "bme680_defs.h"
+#include "bme680_hal.h"
+
+#ifdef MODULE_PERIPH_SPI
+#include "periph/spi.h"
+#endif
+
+#ifdef MODULE_PERIPH_I2C
+#include "periph/i2c.h"
+#endif
+
+#include "xtimer.h"
+
+#define BME680_DEV dev.dev
+
+
+int main(void)
+{
+    bme680_t dev;
+
+    /* You may assign a chip select identifier to be handled later */
+    dev.dev.dev_id = 0;
+    xtimer_sleep(5);
+    /* amb_temp can be set to 25 prior to configuring the gas sensor 
+     * or by performing a few temperature readings without operating the gas sensor.
+     */
+    BME680_DEV.amb_temp = 25;
+    BME680_DEV.delay_ms = ms_sleep;
+
+    printf("Intialize BME680 sensor...");
+    if(bme680_init(&dev, &bme680_params[0]) != BME680_OK) {
+        puts("failed");
+    }
+    else {
+        puts("OK");
+    }
+
+    struct bme680_field_data data;
+    uint8_t ret;
+    while(1)
+    {
+        xtimer_sleep(5);
+
+        ret = bme680_get_sensor_data(&data, &(dev.dev));
```

Comment:
I'm wondering whether it makes really sense to expose the internal API of the BME680 vendor driver or whether it would be better to encapsulate these functions in `static inline functions`. My concern is, that the approach of calling these functions differs from usual driver APIs in RIOT. Usually, driver APIs in RIOT define
```
int xyz_func_abc(xyz_t *dev, int param1, int param2);
```
The `dev` parameter points to the device structure that contain all information to execute it.

```suggestion
        ret = bme680_read(&dev, &data);
```
where `bme680_read` would be defined as
```
static inline int  bme680_read(bme680_t *dev, struct bme680_field_data *data)
{
    assert(dev);
    return bme680_get_sensor_data(data, &(dev->dev));
}
```
Most other internal functions have not be exposed since RIOT design rules prescribe that a sensor is functional after startup and the mode or other parameters of the sensor are defined by `driver_params` during initialization and are not changed afterwards.